compiler/rustc_mir_transform/src/inline/cycle.rs

```diff
@@ -162,5 +165,7 @@ crate fn mir_inliner_callees<'tcx>(
             calls.insert(call);
         }
     }
-    tcx.arena.alloc_from_iter(calls.iter().copied())
+    let mut calls: Vec<_> = calls.into_iter().collect();
+    calls.sort();
```

Comment:
Maybe use FxIndexSet instead of SsoHashSet to obtain the deterministic order? The sort is using transitively Ord for DefId which we are trying to remove.